### DISCUSSION TITLE
after update 1.15.2 to 1.16.9 ; server error 500 for some time records only

### DISCUSSION THREAD

**Discussion by yroberttissot** [kimai/kimai#3053] [On-Premise: Installation & Updates] [answered]
#### Expected behaviour

To be able to open all time records and edit them

####Actual behaviour

If I open any time records from one specific customer, I get the Error 500
Before the update, I had no issues with that customer

After a while, I can't open the timesheet at all. If I reload through the console, I can access again the timesheet, even open the record of the specific client, but if I try to edit and save -> 500 error, cannot save change

And after that, I get the Server Error when I try to open the record

EDIT: even opening some project does the Error 500

####Kimai version [for example: dev 1.1.0.1389|github master|github develop]

1.16.9 prod
PHP 7.4.26
server Linux h2web234 4.9.0-0.bpo.12-amd64 #1 SMP Debian 4.9.210-1+deb9u1~deb8u1 (2020-06-09) x86_64

####Relevant log entries from log file (debug tab or temporary/logfile.txt)

log file is weirdly empty and all permissions look fine.

from the server, I only have these logs
kimai.XXX.com [Tue Jan 04 09:19:33.297743 2022] [fastcgi:error] [pid 22760:tid 140075375515392] (104)Connection reset by peer: [client 194.XXX:0] FastCGI: comm with server "/home/clients/d3ba84e1add40dfe16a751e5ab76e7c7/.config/apache/kimai.XXX.com/.fpm/php5.external" aborted: read failed: "Connection reset by peer", referer https://kimai.XXX.com/public/en/timesheet/
kimai.XXX.com [Tue Jan 04 09:19:33.297814 2022] [fastcgi:error] [pid 22760:tid 140075375515392] [client 194.XXX:0] FastCGI: incomplete headers (0 bytes) received from server "/home/clients/d3ba84e1add40dfe16a751e5ab76e7c7/.config/apache/kimai.XXX.com/.fpm/php5.external", referer https://kimai.XXX.com/public/en/timesheet/

Composer code-check:
sh: vendor/bin/php-cs-fixer: No such file or directory
Script vendor/bin/php-cs-fixer fix --dry-run --verbose --show-progress=none handling the codestyle event returned with error code 127
Script @codestyle was called via kimai:codestyle
Script @kimai:codestyle was called via kimai:pre-commit
Script @kimai:pre-commit was called via code-check

composer tests:
> vendor/bin/phpunit tests/
sh: vendor/bin/phpunit: No such file or directory
Script vendor/bin/phpunit tests/ handling the tests event returned with error code 127

**Comment by kevinpapst** [comment]
Did you check the Troubleshooting section (scroll down) at https://www.kimai.org/documentation/updates.html

**Reply by yroberttissot** [reply]
yeah, i had to do a git stash, then could get the newest version.

Now:
HEAD detached at 1.16.9
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        modified:   symfony.lock

no changes added to commit (use "git add" and/or "git commit -a")

**Reply by yroberttissot** [reply]
update: I have the same issue with 1.16.10

again the error logs
1. FastCGI: incomplete headers (0 bytes) received from server "/home/clients/d3ba84e1add40dfe16a751e5ab76e7c7/.config/apache/XXX.inoshift.com/.fpm/php5.external"
2. )Connection reset by peer: [client 194.XXX:0] FastCGI: comm with server "/home/clients/d3ba84e1add40dfe16a751e5ab76e7c7/.config/apache/kimai.XXX.com/.fpm/php5.external" aborted: read failed: "Connection reset by peer"

any ideas how I can solve this?

PHP version is 7.4.26. all required PHP extension are green in Kimai

**Reply by kevinpapst** [reply]
No idea. 1.16 works in hundreds of installations (nginx) and I have customers that use Apache as well. 
Never saw that error, but I have also never seen apache with FPM. Have you tried using mod php instead?

**Comment by yroberttissot** [comment, marked as answer]
problem was solved by updating PHP 7.4.26 to 8.0.13
but not sure what the exact issue was. there was no problem with Kimai 1.15 and php 7.4

**Reply by kevinpapst** [reply]
Really weird. Some dependent package must have triggered a PHP bug. 

The problem is that I cannot update all packages to their latest version, due to the fact that PHP 7 is still supported. This will change with 1.18, but it is good that you are now already on PHP 8 and that the problem is solved.
